database/src/ops/block.rs

```diff
@@ -1,89 +1,398 @@
+}
+
+/// TODO
+///
+/// # Errors
+/// TODO
+#[inline]
+pub fn get_block_v1<'env, Ro, Rw, Env>(
+    env: &Env,
+    tx_ro: &Ro,
+    height: BlockHeight,
+) -> Result<BlockInfoV1, RuntimeError>
+where
+    Ro: TxRo<'env>,
+    Rw: TxRw<'env>,
+    Env: EnvInner<'env, Ro, Rw>,
+{
+    env.open_db_ro::<BlockInfoV1s>(tx_ro)?.get(&height)
+}
+
+/// TODO
+///
+/// # Errors
+/// TODO
+#[inline]
+pub fn get_block_v2<'env, Ro, Rw, Env>(
+    env: &Env,
+    tx_ro: &Ro,
+    height: BlockHeight,
+) -> Result<BlockInfoV2, RuntimeError>
+where
+    Ro: TxRo<'env>,
+    Rw: TxRw<'env>,
+    Env: EnvInner<'env, Ro, Rw>,
+{
+    env.open_db_ro::<BlockInfoV2s>(tx_ro)?.get(&height)
+}
+
+/// TODO
+///
+/// # Errors
+/// TODO
+#[inline]
+pub fn get_block_v3<'env, Ro, Rw, Env>(
+    env: &Env,
+    tx_ro: &Ro,
+    height: BlockHeight,
+) -> Result<BlockInfoV3, RuntimeError>
+where
+    Ro: TxRo<'env>,
+    Rw: TxRw<'env>,
+    Env: EnvInner<'env, Ro, Rw>,
+{
+    env.open_db_ro::<BlockInfoV3s>(tx_ro)?.get(&height)
+}
+
+/// TODO
+///
+/// # Errors
+/// TODO
+#[inline]
+pub fn get_top_block<'env, Ro, Rw, Env>(
+    env: &Env,
+    tx_ro: &Ro,
+) -> Result<(BlockHeight, BlockInfoLatest), RuntimeError>
+where
+    Ro: TxRo<'env>,
+    Rw: TxRw<'env>,
+    Env: EnvInner<'env, Ro, Rw>,
+{
+    env.open_db_ro::<BlockInfoV3s>(tx_ro)?.last()
+}
+
+/// TODO
+///
+/// # Errors
+/// TODO
+#[inline]
+pub fn get_block_height<'env, Ro, Rw, Env>(
+    env: &Env,
+    tx_ro: &Ro,
+    block_hash: &BlockHash,
+) -> Result<BlockHeight, RuntimeError>
+where
+    Ro: TxRo<'env>,
+    Rw: TxRw<'env>,
+    Env: EnvInner<'env, Ro, Rw>,
+{
+    env.open_db_ro::<BlockHeights>(tx_ro)?.get(block_hash)
+}
+
+/// TODO
+///
+/// # Errors
+/// TODO
+#[inline]
+pub fn block_exists<'env, Ro, Rw, Env>(
+    env: &Env,
+    tx_ro: &Ro,
+    block_hash: &BlockHash,
+) -> Result<bool, RuntimeError>
+where
+    Ro: TxRo<'env>,
+    Rw: TxRw<'env>,
+    Env: EnvInner<'env, Ro, Rw>,
+{
+    env.open_db_ro::<BlockHeights>(tx_ro)?.contains(block_hash)
+}
+
+/// TODO
+pub fn get_block_header() {
     todo!()
 }
 
 /// TODO
-pub fn get_block_long_term_weight() {
+pub fn get_block_header_from_height() {
     todo!()
 }
 
 /// TODO
-pub fn get_block_timestamp() {
+pub fn get_top_block_hash() {
     todo!()
 }
 
 /// TODO
-pub fn get_block_cumulative_rct_outputs() {
+pub fn get_block_hash() {
     todo!()
 }
 
 /// TODO
-pub fn get_block() {
+pub fn get_block_weight() {
     todo!()
 }
 
 /// TODO
-pub fn get_block_from_height() {
+pub fn get_block_already_generated_coins() {
     todo!()
 }
 
 /// TODO
-pub fn get_block_header() {
+pub fn get_block_long_term_weight() {
     todo!()
 }
 
 /// TODO
-pub fn get_block_header_from_height() {
+pub fn get_block_timestamp() {
     todo!()
 }
 
 /// TODO
-pub fn get_top_block() {
+pub fn get_block_cumulative_rct_outputs() {
     todo!()
 }
 
 /// TODO
-pub fn get_top_block_hash() {
+pub fn get_block_from_height() {
     todo!()
 }
```

Comment:
These functions that are basically `get_block()` -> return some field - should they exist?

There is already `get_block()`, users can just `get_block()?.field` right?